src/ng/filter/orderBy.js

```diff
@@ -176,88 +176,110 @@
 orderByFilter.$inject = ['$parse'];
 function orderByFilter($parse) {
   return function(array, sortPredicate, reverseOrder) {
+
     if (!(isArrayLike(array))) return array;
-    sortPredicate = isArray(sortPredicate) ? sortPredicate : [sortPredicate];
+
+    if (!isArray(sortPredicate)) { sortPredicate = [sortPredicate]; }
     if (sortPredicate.length === 0) { sortPredicate = ['+']; }
-    sortPredicate = sortPredicate.map(function(predicate) {
-      var descending = false, get = predicate || identity;
-      if (isString(predicate)) {
+
+    var predicates = processPredicates(sortPredicate);
+    var compareValues = Array.prototype.map.call(array, getComparisonObject);
+
+    compareValues.sort(doComparison);
+    array = compareValues.map(function(item) { return item.value; });
+    if (reverseOrder) { array.reverse(); }
```

Comment:
It could but if you look at the previous implementation it meant complicating the processing of the predicate functions, which makes it significantly more messy.
What is the cost of `Array.reverse()`? It would seem to me that it could be done rather quickly inside the JavaScript engine. Any thoughts @caitp ?